apps/omg_child_chain/lib/omg_child_chain/fees/json_fee_parser.ex

```diff
@@ -29,21 +29,23 @@ defmodule OMG.ChildChain.Fees.JSONFeeParser do
           | :invalid_tx_type
 
   @doc """
-  Parses and validates json encoded fee specifications file
+  Parses and validates json encoded fee specifications response
   Parses provided json string to token-fee map and returns the map together with possible parsing errors
   """
-  @spec parse(binary()) ::
+  @spec parse(binary() | map() | list()) ::
           {:ok, OMG.Fees.full_fee_t()}
           | {:error, list({:error, parsing_error(), any(), non_neg_integer() | nil})}
-  def parse(file_content) do
-    with {:ok, json} <- Jason.decode(file_content) do
-      {errors, fee_specs} = Enum.reduce(json, {[], %{}}, &reduce_json/2)
-
-      errors
-      |> Enum.reverse()
-      |> (&{&1, fee_specs}).()
-      |> handle_parser_output()
-    end
+  def parse(file_content) when is_binary(file_content) do
+    with {:ok, json} <- Jason.decode(file_content), do: parse(json)
```

Comment:
whats with this `with`? remove it and crash

Reply:
So it's again the way we implement the FeeServer with Adapters, is that it's FeeServer handles the errors. It's not left to the Parser or Adapter's decision

Reply:
ahhhaaa, so this would bubble all the way up to `Alarm.set(Alarm.invalid_fee_source(__MODULE__))`! cool!